Anime-App/Feature/Anime/View/ViewControllers/AnimeViewController.swift

```diff
@@ -117,59 +124,93 @@ extension AnimeViewController {
         createSearchBar()
         view.addSubview(tableView)
         self.tableView.addSubview(self.refreshControl)
+        
+        customLoadingView.center = self.view.center
+        self.view.addSubview(customLoadingView)
+        customLoadingView.bringSubviewToFront(self.view)
+        
         spinner.center = self.view.center
         self.view.addSubview(spinner)
         spinner.bringSubviewToFront(self.view)
-
+        
         NSLayoutConstraint.activate([
+            
+            customLoadingView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            customLoadingView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            
             spinner.centerXAnchor.constraint(equalTo: view.centerXAnchor),
-            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: 50),
+            
             tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
             tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 16),
             tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -16),
             tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
         ])
     }
     
-    func fetchAnime() {
-        self.isFetching = true
-        self.spinner.startAnimating()
+    
+    
+    
+    func resetAnime() {
+        animeViewModel.resetAnime { res in
+            switch res {
+            case .success:
+                self.tableView.reloadData()
+            case .failure:
+                print("Failed to reset")
+            }
+        }
+    }
+    
+    func startPagination() {
+        if isPaginating && !isFetching {
+             self.spinner.startAnimating()
+        }
+        isPaginating = true
         animeViewModel.fetchAnime(page: page)  { [weak self] res in
             switch res {
             case .success:
                 self?.tableView.reloadData()
-             case .failure:
+                self?.page+=1
+            case .failure:
                 print("no more data")
-             }
+            }
+            self?.isPaginating = false
             self?.spinner.stopAnimating()
-            self?.isFetching = false
         }
     }
     
-    func resetAnime() {
-        animeViewModel.resetAnime { res in
+    
+    func fetchAnime() {
+        if !isFetching && !isPaginating {
+            self.customLoadingView.startAnimating()
+        }
+         UIView.animate(withDuration: 1, delay: 0.25, usingSpringWithDamping: 0.5, initialSpringVelocity: 5, options: .curveEaseInOut, animations: {
```

Comment:
This is still wrong tobi. You don't want to add in a hard coded delay for when you show and hide the spinner. This should be in the closure when everything has finished. And if you do this then there is no need for a delay. 

When you work with api's the way you want to tackle updating UI when the API call has finished is directly in the closure block so anything that you want to do when the API call has finished is done at the very end in the closure block so can you review the rest of this file and also change it in other places as well as moving this animation code into the closure block at the end as the last thing